### PR TITLE
Use the shortcode pool for Notify SMS

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Use the shortcode pool for GC Notify's SMS.

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/377

# Test instructions | Instructions pour tester la modification

Send SMS from the GC Notify service. Should be from the shortcode pool.

# Release Instructions | Instructions pour le déploiement

None.

# Reviewer checklist | Liste de vérification du réviseur

- [x] This PR does not break existing functionality.
- [x] This PR does not violate GCNotify's privacy policies.
- [x] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [x] This PR does not significantly alter performance.
- [x] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: app/clients/sms/aws_pinpoint.py [hunk 1]
```diff
@@ -21,13 +21,17 @@ def init_app(self, current_app, statsd_client, *args, **kwargs):
...
+    def send_sms(self, to, content, reference, multi=True, sender=None, template_id=None, service_id=None):
         messageType = "TRANSACTIONAL"
         matched = False
         opted_out = False
         response = {}
 
-        if template_id is not None and str(template_id) in self.current_app.config["AWS_PINPOINT_SC_TEMPLATE_IDS"]:
+        use_shortcode_pool = (
+            str(template_id) in self.current_app.config["AWS_PINPOINT_SC_TEMPLATE_IDS"]
+            or str(service_id) == self.current_app.config["NOTIFY_SERVICE_ID"]
```
Comment (ben851) [inline app/clients/sms/aws_pinpoint.py:32]: Why don't we just put the notify service id in the AWS_PINPOINT_SC_TEMPLATE_IDS?

Comment (sastels) [reply app/clients/sms/aws_pinpoint.py:32]: We could but we plan to move away from this env var entirely and have the pool choice be more dynamic. The `NOTIFY_SERVICE_ID` variable already exists so this way we don't need to change `AWS_PINPOINT_SC_TEMPLATE_IDS` and can eventually remove it.

**Review by ben851 — APPROVED** [review]
LGTM

**Review by jimleroyer — APPROVED** [review]
LGTM!

**Pull request merged**